src/Estimators/SelfHealingOverlap.h

```diff
@@ -33,13 +33,28 @@ class SelfHealingOverlap : public OperatorEstBase
   using ValueType   = QMCTraits::ValueType;
   using PosType     = QMCTraits::PosType;
 
+
+  enum wf_types
+  {
+    msd_wf = 0,
+    sd_rot_wf,
+    no_wf
+  };
+
   //data members set only during construction
   const SelfHealingOverlapInput input_;
 
   /** @ingroup SelfHealingOverlap mutable data members
    */
   Vector<ValueType> det_ratios;
 
+  /// wavefunction type
+  wf_types wf_type;
+
+  /// use direct parameter derivative for MSD or not
+  bool use_param_deriv;
```

Comment:
Please add const. It can be decided in the constructor.

Reply:
Adding const makes this line scream in the constructor:

use_param_deriv = inp.get<bool>("param_deriv");

(constructor is based on input section like other estimators)

See e.g. MomentumDistribution.* and setting member my_name_ from the OperatorEstBase class.

Reply:
I fixed it. Just move it to member initializer lists of the constructor.